### ISSUE TITLE
Not able to derive rules as expected

### ISSUE THREAD

**Issue by jeswr** [eyereasoner/eye#79]
If I have 

```
@prefix : <http://example.org/> .

{ [] :rule ?O } => ?O .
:s :p [ :p1 :o ] .
{ :s!:p :p1 :o . } => { :p :q :r . } .
```

then I derive `:p :q :r` - nothing exciting here.

Now if I have

```
@prefix : <http://example.org/> .

{ [] :rule ?O } => ?O .
:s :p [ :p1 :o ] .
: :rule { { :s!:p :p1 :o . } => { :p :q :r . } . } .
```

Then `:p :q :r` is not derived; even though `{:s :p _:bn_3. _:bn_3 :p1 :o} => {:p :q :r}.` is.

---

Note that everything works as expected without this path sequence; that is

```
@prefix : <http://example.org/> .

{ [] :rule ?O } => ?O .
:s :p :o .
: :rule { { :s :p :o . } => { :p :q :r . } . } .
```

derives `:p :q :r`

**Comment by josd** [comment]
Thanks for your observation and it should now be fixed in EYE v3.4.8

**Comment by jeswr** [comment]
Thanks!
